src/commands/play.rs

```diff
@@ -276,9 +276,10 @@ async fn enqueue_song(
     query_type: QueryType,
 ) {
     let source_return = match query_type {
-        QueryType::VideoLink => YouTubeRestartable::ytdl(query, true).await,
+        QueryType::VideoLink | QueryType::PlaylistLink => {
```

Comment:
In the context of an `enqueue_song` how can a Playlist link get here? The `enqueue_ playlist` handles playlist links, not `enqueue_song`. My point being, this does not seem the correct way to fix it (even tho it does fix it)

Reply:
This fix doesn't seem right, it should remain unreachable because it doesn't make sense for `enqueue_song` to receive playlist links. Given that the output of the `YouTubeRestartable::ytdl_playlist` always is an array of links, the fix should be passing `QueryType::VideoLink` to the `enqueue_song` method inside the loop.

Can you test this?

Reply:
> I think we should just hardcode pass as a parameter the query type to VideoLink right? Is there any case where ytdl won't return a single link/url ?

This was the other approach that I considered, and if you think it fits more, I'll make such a change.